lib/board.rb

```diff
@@ -0,0 +1,65 @@
+class Board
+  attr_accessor :cells, :turn_count, :exes, :oes
+
+  def initialize
+    @cells = Array.new(9, " ")
+    @exes = []
+    @oes = []
+  end
+
+  def reset!
+    @cells = Array.new(9, " ")
+  end
+
+  def display
+    puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+    puts "-----------"
+    puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+    puts "-----------"
+    puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+  end
+
+  def position(input)
+    index = input.to_i - 1
+    @cells[index]
+  end
+
+  def exes
+    @exes = cells.map.with_index.map { |mark, i| i if mark == "X" }
+  end
+
+  def oes
+    @oes = cells.map.with_index.map { |mark, i| i if mark == "O" }
+  end
+
+  def full?
+    return false if @cells.any? { |e| e == " " }
+    true
+  end
+
+  def taken?(pos)
+    self.position(pos) != " "
+  end
+
+  def turn_count
+    turn_count = 0
+    @cells.each { |e| turn_count += 1 if e != " " }
+    turn_count
+  end
+
+  def valid_move?(move)
+    if self.taken?(move) || !((1..9).include?(move.to_i))
```

Comment:
Because `#taken?` and `.include?` return boolean values implicitly, you don't have to return true or false. 

```
!((1..9).include?(move.to_i) && !taken(input)
```